File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int prefix = 1;
        int[] output = new int[nums.length];

        for (int i = 0; i < nums.length; i++) {
            output[i] = prefix;
            prefix *= nums[i];
        }

        prefix = 1;

        for (int i = nums.length - 1; i >= 0; i--) {
            output[i] *= prefix;
            prefix *= nums[i];
        }

        return output;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        # Need to track indexes for each num
        idxTrack = {}

        # Then loop over the numbers 
        for idx in range(0, len(nums)):
            # We have to initialize the keys to an empty list
            lst = idxTrack.setdefault(nums[idx], [])

            # Add idx to current list
            lst.append(idx)

            # Get complement
            comp = target - nums[idx]

            # Check for complement in track, and comIdx != idx 
            compLst = idxTrack.get(comp)

            if not compLst:
                continue
            
            if compLst[0] != idx:
                return [compLst[0], idx]

        return []